packages/wrangler/templates/pages-template-plugin.ts

```diff
@@ -133,15 +133,19 @@ export default function (pluginArgs: unknown) {
 		const handlerIterator = executeRequest(request, relativePathname);
 		const pluginNext = async (input?: RequestInfo, init?: RequestInit) => {
 			if (input !== undefined) {
-				request = new Request(input, init);
+				let url = input;
+				if (typeof input === "string") {
+					url = new URL(input, request.url).toString();
+				}
+				request = new Request(url, init);
 			}
 
 			const result = handlerIterator.next();
 			// Note we can't use `!result.done` because this doesn't narrow to the correct type
 			if (result.done === false) {
 				const { handler, params, path } = result.value;
 				const context = {
-					request,
+					request: new Request(request.clone()),
```

Comment:
Why can't we just say: `request: request.clone()`?

Reply:
https://cloudflareworkers.com/#62860051d80fa1cbe10b76042242694f:https://tutorial.cloudflareworkers.com